--- python/PySide6/__init__.py	2025-01-11 15:01:54.266726602 -0500
+++ python/PySide6/__init__.py	2025-01-11 15:06:19.646365323 -0500
@@ -10,14 +10,9 @@
 
 
 def _additional_dll_directories(package_dir):
-    # Find shiboken6 relative to the package directory.
-    root = os.path.dirname(package_dir)
-    # Check for a flat .zip as deployed by cx_free(PYSIDE-1257)
-    if root.endswith('.zip'):
-        return []
-    shiboken6 = os.path.join(root, 'shiboken6')
-    if os.path.isdir(shiboken6): # Standard case, only shiboken6 is needed
-        return [shiboken6]
+    # rez-pip patch: Return the path to the shiboken rez package
+    return [os.path.join(os.environ["REZ_SHIBOKEN6_ROOT"], "python", "shiboken6")]
+
     # The below code is for the build process when generate_pyi.py
     # is executed in the build directory. We need libpyside and Qt in addition.
     shiboken6 = os.path.join(os.path.dirname(root), 'shiboken6', 'libshiboken')
